### PR TITLE
Fixed ParseBytes and ParseFile by adding `agclearerrors` in C

### PR DESCRIPTION
**Pull request by denk0403**
A fix for the bug described by issue #60.

In summary, this PR introduces `agclearerrors()` in the C code, which closes and resets the output file for error reporting. In the `ccall` package, a binding is created via `Agclearerrors()` function.

`ParseBytes()` and `ParseFile()` in the `cgraph` package now call `Agclearerrors()` before reading and parsing the buffer to ensure there are no extraneous errors left over, and that the call to `Aglasterr()` returns only new errors produced by parsing.

### CODE REVIEW

**Comment by goccy** [comment]
Thank you for your contribution ! I've commented

**Comment by denk0403** [comment]
The tests should pass for Go version 1.15 now!

File: internal/ccall/cgraph/agerror.c [hunk 1]
```diff
@@ -169,10 +169,17 @@ void agwarningf(const char *fmt, ...)
...
-{ 
+int agreseterrors()
+{
     int rc = agmaxerr;
     agmaxerr = 0;
-    return rc; 
+    return rc;
 }
 
+void agclearerrors()
```
[the remark this answers is not in the document]
Comment (denk0403) [reply internal/ccall/cgraph/agerror.c:179]: `agclearerrors()` is not present in the new `agerror.c`, it is an original function. I understand that means this version of the library now differs from the source of truth. However, as far as I can tell, the library provides no way of resetting the `agerrout` file or clearing the last error. 

The most one can do is set it to a different string via `agerr()`, but the Go bindings currently expect `nil` to denote the absence of an error. So I strongly believe this function is necessary.

In order to not modify the C files, I considered defining it at the top of [`internal/ccall/cgraph.go`](https://github.com/goccy/go-graphviz/blob/master/internal/ccall/cgraph.go) where it seems there is already custom C code, but `cgraph.h` doesn't expose the necessary global variables to get it to work anyways.

[the remark this answers is not in the document]
Comment (goccy) [reply internal/ccall/cgraph/agerror.c:179]: I understand, LGTM ! Thank you :)

**Pull request merged**